runtime/src/identity.rs

```diff
@@ -589,6 +641,12 @@ impl<T: Trait> Module<T> {
         return false;
     }
 
+    /// It returns `true` if `curr_id`'s master key is a signing_key at `target_id`.
+    pub fn is_authorized_identity(curr_id: IdentityId, target_id: IdentityId) -> bool {
+        let curr_master_key = Self::did_records(curr_id).master_key;
+        Self::is_authorized_key(target_id, &curr_master_key)
```

Comment:
Since a key can either be a master key or a signing key, this will never return true. We need to allow dids to be added as signing key(or signing dids) themselves. Another requirement is that if did A is signing did of did B then not only the master key but all signing keys of did A should be allowed to act on behalf of did B (TODO if they have proper permission).

Reply:
I've added initial API to add/remove signing identities: 
- `Identity::add_signing_identities` and,
- `Identity::remove_signing_identites`.